proxy/src/registry.rs

```diff
@@ -375,7 +375,7 @@ impl Client for Registry {
         let block = self.client.block_header(applied.block).await?;
         let tx = Transaction::confirmed(
             Hash(applied.tx_hash),
-            block.number,
+            block.map(|b| b.number),
```

Comment:
Maybe `block` could have a helper `to_number: Block -> Option<Number>`

Reply:
The problem is that it's `block` that is `Option`al.